### ISSUE TITLE
plt.tight_layout(pad=0) cuts away outer ticks

### ISSUE THREAD

**Issue by MaxNoe** [matplotlib/matplotlib#5281]
Wenn I do `plt.tight_layout(pad=0)` outer ticks are not considered and cut away:

```{python}
import matplotlib.pyplot as plt

fig, ax = plt.subplots()

ax.get_yaxis().set_tick_params(direction='out')
ax.get_xaxis().set_tick_params(direction='out')
ax.set_xlim(0.05, 0.95)

fig.tight_layout(pad=0)
fig.savefig('outer_ticks.png', dpi=300)
```

Result:

![outer_ticks](https://cloud.githubusercontent.com/assets/5488440/10606087/9b30e862-7730-11e5-9772-8d80326b5c4b.png)

**Comment by mdboom** [comment]
Milestoning this as 2.0, since if we choose to go with outer ticks in the new defaults, we'll need to first fix this bug.

**Comment by mdboom** [comment]
Fixed by #5683.
